django/db/models/query.py

```diff
@@ -321,6 +321,7 @@ def iterator(self):
         An iterator over the results from applying this QuerySet to the
         database.
         """
+        self.query.chunked_fetch = True
```

Comment:
I am not sure I understand your last point. When the reference to the `QuerySet` falls out of scope, the last reference to the cursor is [finally closed](https://github.com/django/django/blob/master/django/db/models/sql/compiler.py#L1224-L1225), even when the iterator is not completely exhausted. Or maybe I have missed something?

I did the following experiment, it would seem to indicate that the finally clause is called even when the iterator has not been fully consumed.
```python
class Cursor():
    def __del__(self):
        print('Cursor %r deleted' % self)

def cursor_iter():
    try:
        cur = Cursor()
        while True:
            yield cur
    finally:
        print('End of generator, deleting cursor')

def user():
    c = cursor_iter()
    for _ in range(3):
        next(c)
    # c is not totally consumed and falls out of scope

user()
```
prints:
```
End of generator, deleting cursor
Cursor <__main__.Cursor instance at 0x7fe840846758> deleted
```